src/lascopcindex.cpp

```diff
@@ -0,0 +1,1541 @@
+      }
+
+      if (lasreader->header.version_minor < 4)
+      {
+        lasreader->header.version_minor = 4;
+        lasreader->header.extended_number_of_point_records = lasreader->header.number_of_point_records;
+        lasreader->header.number_of_point_records = 0;
+        for (i = 0; i < 5; i++)
+        {
+          lasreader->header.extended_number_of_points_by_return[i] = lasreader->header.number_of_points_by_return[i];
+          lasreader->header.number_of_points_by_return[i] = 0;
+        }
+      }
+
+      // Upgrade to point format > 5
+      if (lasreader->header.point_data_format < 6 || lasreader->header.point_data_format > 8)
+      {
+        // were there extra bytes before
+        I32 num_extra_bytes = 0;
+        switch (lasreader->header.point_data_format)
+        {
+          case 0: num_extra_bytes = lasreader->header.point_data_record_length - 20; break;
+          case 1: num_extra_bytes = lasreader->header.point_data_record_length - 28; break;
+          case 2: num_extra_bytes = lasreader->header.point_data_record_length - 26; break;
+          case 3: num_extra_bytes = lasreader->header.point_data_record_length - 34; break;
+          case 4: num_extra_bytes = lasreader->header.point_data_record_length - 57; break;
+          case 5: num_extra_bytes = lasreader->header.point_data_record_length - 63; break;
+          case 6: num_extra_bytes = lasreader->header.point_data_record_length - 30; break;
+          case 7: num_extra_bytes = lasreader->header.point_data_record_length - 36; break;
+          case 8: num_extra_bytes = lasreader->header.point_data_record_length - 38; break;
+          case 9: num_extra_bytes = lasreader->header.point_data_record_length - 59; break;
+          case 10: num_extra_bytes = lasreader->header.point_data_record_length - 67; break;
+        }
+        if (num_extra_bytes < 0)
+        {
+          fprintf(stderr, "ERROR: point record length has %d fewer bytes than needed\n", num_extra_bytes);
+          byebye(true);
+        }
+
+        lasreader->header.clean_laszip();
+        lasreader->header.point_data_format = (U8)target_point_data_format;
+        switch (lasreader->header.point_data_format)
+        {
+          case 6: lasreader->header.point_data_record_length = 30 + num_extra_bytes; break;
+          case 7: lasreader->header.point_data_record_length = 36 + num_extra_bytes; break;
+          case 8: lasreader->header.point_data_record_length = 38 + num_extra_bytes; break;
+        }
+      }
+
+      // convert CRS from GeoTIF to OGC WKT
+      if (lasreader->header.vlr_geo_keys)
+      {
+        char* ogc_wkt = 0;
+        I32 len = 0;
+        geoprojectionconverter.set_projection_from_geo_keys(lasheader->vlr_geo_keys[0].number_of_keys, (GeoProjectionGeoKeys*)lasheader->vlr_geo_key_entries, lasheader->vlr_geo_ascii_params, lasheader->vlr_geo_double_params);
+        if (geoprojectionconverter.get_ogc_wkt_from_projection(len, &ogc_wkt))
```

Comment:
Wont this write out the error when it does convert to WKT, rather than when it doesn't?

Reply:
Sure, I fixed it. Thank you for catching it.